src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -358,12 +358,9 @@ public static FormulaValue Shuffle(IRContext irContext, FormulaValue[] args)
             var table = (TableValue)args[0];
             var records = table.Rows;
 
-            lock (_randomizerLock)
-            {
-                _random ??= new Random();
-            }
+            var random = _defaultRandService;
```

Comment:
Same comment as `RandBetween`: need to attempt to retrieve the service from the runner